src/lib/mdns/AllInterfacesListenIterator.h

```diff
@@ -0,0 +1,128 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdio.h>
+
+#include <mdns/minimal/Server.h>
+
+namespace chip {
+namespace Mdns {
+
+/// Checks if the current interface is powered on
+/// and not local loopback.
+template <typename T>
+bool IsCurrentInterfaceUsable(T & iterator)
+{
+    if (!iterator.IsUp() || !iterator.SupportsMulticast())
+    {
+        return false; // not a usable interface
+    }
+    char name[chip::Inet::InterfaceIterator::kMaxIfNameLength];
+    if (iterator.GetInterfaceName(name, sizeof(name)) != CHIP_NO_ERROR)
+    {
+        ChipLogError(Discovery, "Failed to get interface name.");
+        return false;
+    }
+
+    // TODO: need a better way to ignore local loopback interfaces/addresses
+    // We do not want to listen on local loopback even though they are up and
+    // support multicast
+    //
+    // Some way to detect 'is local looback' that is smarter (e.g. at least
+    // strict string compare on linux instead of substring) would be better.
+    //
+    // This would reject likely valid interfaces like 'lollipop' or 'lostinspace'
+    if (strncmp(name, "lo", 2) == 0)
```

Comment:
on Mac we also get errors on some of the other interfaces. Are you running on linux only?

Reply:
This is actually just moved from another spot - it's not new code. In the new commit, I'm pushing the commissionable node discovery through the resolver, and mac will continue to use the platform implementation.